packages/openapi/src/openapi-request-builder.ts

```diff
@@ -164,23 +164,26 @@ export class OpenApiRequestBuilder<ResponseT = any> {
 
   private getPath(): string {
     const pathParameters = this.parameters?.pathParameters || {};
+    let path = this.pathPattern;
 
-    const pathParts = this.pathPattern.split('/');
-    return pathParts
-      .map(part => {
-        if (OpenApiRequestBuilder.isPlaceholder(part)) {
-          const paramName = part.slice(1, -1);
-          const paramValue = pathParameters[paramName];
-          if (!paramValue) {
-            throw new Error(
-              `Cannot execute request, no path parameter provided for '${paramName}'.`
-            );
-          }
-          return encodeURIComponent(paramValue);
-        }
-        return part;
-      })
-      .join('/');
+    // Get the innermost curly bracket pairs with non-empty and legal content as placeholders.
+    const placeholders = path.match(/{[^/?#{}]+}/g);
+    if (!placeholders) {
+      return path;
+    }
+
+    placeholders.map(placeholder => {
+      const strippedPlaceholder = placeholder.slice(1, -1);
+      const parameterValue = pathParameters[strippedPlaceholder];
+      if (/[/?#]+/.test(parameterValue)) {
+        throw new Error(
+          `Cannot execute request, the value of the path parameter '${strippedPlaceholder}' must not contain '/', '?', or '#'. (RFC3986)`
+        );
+      }
+      path = path.replace(placeholder, encodeURIComponent(parameterValue));
```

Comment:
Do we need the check this if we encode the component afterwards? That's exactly why we're doing it, isn't it?

Reply:
Oh, I guess you are right. I somehow thought that these characters won't be encoded but I was wrong after a double-check. Thanks for the input.